docs/DeveloperGuide.md

```diff
@@ -280,6 +247,46 @@ and that minimal calculation is needed to reset the bill to 0 (`b/-CURRENT_VALUE
 **Aspect: Using the `b/` prefix**
 * To standardise the formatting and testing for field inputs, we decided to include the `b/` prefix in the command syntax.
 
+### DateRange Field
+
+#### Implementation:
+
+* The `DateRange` class holds the period of stay of a `Guest`.
+* Its constructor takes in a string representing a check-in date and a check-out date.
+
+#### Design Considerations:
+
+**Aspect: How to represent dates**
+* **Alternative 1:** Separate classes for check-in and check-out dates extended from a `GuestDate` class.
+    * Pros: Easier to parse and edit dates separately.
+    * Cons: Depend on each other for validation (check-in must be earlier than check-out), which increases coupling.
+* **Alternative 2 (current choice):** Single class representing both dates.
+    * Pros: Validation can be done within the class itself which reduces coupling. More intuitive as the dates
+      are often displayed together.
+    * Cons: Parsing, editing, and other operations on the dates are more complex.
+
+Taking into consideration that check-in and check-out dates come as a pair, we decided to proceed with Alternative 2 to reduce coupling.
+
+### Edit Command
+
+#### Implementation:
+* The `edit` command takes in an INDEX indicating the index of the guest to edit in the current panel (starting from 1)
+  and 8 optional fields (`Name`, `Phone`, `Email`, `Room`, `Date Range`, `Number Of Guests`, `Is Room Clean` and `Request`)
+  and is supported by the `EditCommandParser` that extracts out each of the fields from their respective prefixes.
+
+The following activity diagrams summarizes what happens when a user enters an `edit` command.
```

Comment:
Should this be the following activity 'diagram'?

Reply:
yes!